gnucash/report/report-system/report-utilities.scm

```diff
@@ -493,9 +493,17 @@ flawed. see report-utilities.scm. please update reports.")
 (define* (gnc:account-accumulate-at-dates
           acc dates #:key
           (nosplit->elt #f)
-          (split->date (compose xaccTransGetDate xaccSplitGetParent))
+          (split->date #f)
           (split->elt xaccSplitGetBalance))
-  (let lp ((splits (xaccAccountGetSplitList acc))
+  (define to-date
+    (or split->date
+        (compose xaccTransGetDate xaccSplitGetParent)))
+  (define splits
+    (if split->date
+        (stable-sort! (lambda (a b) (< (split->date a) (split->date b)))
```

Comment:
Don't you want `to-date` instead of `split->date` here? It will crash if `split->date` is defaulted.

Reply:
It's fine; if `split->date` is #f then the splitlist won't be sorted at all.